meilisearch-release-changelog/v1.0.0.md

```diff
@@ -0,0 +1,122 @@
+
+<!-- The following lines should ONLY be put in the official release changelogs -->
+🧰 Most integrations (SDKs, clients, libraries) will be compatible with this version within four hours of the release. Sometimes this can take up to 48 hours, depending on the issues we encounter during the release.
+
+<!-- The following lines should ONLY be put in the official release changelogs -->
+Here is the exhaustive list of integrations  **not immediately compatible** with v1.0.0:
+- TBD
+
+<!-- The following line should NOT be put in the official release changelogs -->
+📖 Meilisearch's official documentation will be ready for the official v1.0.0 release. Meanwhile, work-in-progress documentation is available [here]().
+
+# New features and improvements 🔥
+
+# Charabia
+
+TBD
+
+# Generate master key option
+
+You can now take advantage of the `--generate-master-key` CLI option.
+
+By using that option, you can easily generate a strong master key to secure your Meilisearch instance easily. Using that option won't start Meilisearch and will exit without any error.
+
+Done in #3274 by @dureuill
+
+## Other improvements
+
+* Search speed improvements:
+  * Significantly improve the memory usage of search requests containing multiple long words (#3115) @loiclec
+  * Significantly improve the performance of the `exactness` ranking rule for search requests containing many words (#3116) @loiclec
+  * Translate multi-word synonyms into [phrases](https://docs.meilisearch.com/learn/what_is_meilisearch/features.html#phrase-search) when interpreting a search query. This change improves the relevancy of the results and stabilises the latency of search queries containing many multi-word synonyms, removing a source of denial-of-service attacks. (#3125) @loiclec
+  * Improve the performance of the `proximity` ranking rule for search requests ending with short words (#3118) @loiclec
+* You can now migrate from any old versions of Meilisearch (supporting the dump feature) to the latest version using a single dump. We hope to facilitate your migration to v1.0.0 so you can enjoy the latest features as easily as possible (#3170) @dureuill
+* Improve [`download-latest.sh`](https://github.com/meilisearch/meilisearch/blob/main/download-latest.sh) script: integrate apple-silicon binary (#3169) @curquiza
+* Clarify error message when the DB and engine versions are incompatible (#3279) @dureuill
+
+# Breaking changes ⚠️
+
+Since we are releasing v1.0.0, this is the last time we are introducing breaking changes to this release before v2.0.0.
+
+## New CLI behavior
+
+* For security reasons, Meilisearch will now reject master keys that are less than 16 bytes (#3274) @dureuill
+* Remove `--max-index-size` and `--max-task-db` configuration options (#3278) @dureuill
+Indeed, these options did not include the real space Meilisearch takes on the disk and were not adapted to the user's needs.
+* Remove `--disable-auto-batching` CLI option (#3296) @loiclec
```

Comment:
When talking about instance options, should we mention both the environment variable and CLI option (like line57) or just the CLI option?

Reply:
Hum yes you're right, would be clearer!